packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js

```diff
@@ -93,8 +96,21 @@ export default function SidebarNavigationScreenPatterns() {
 	const { templatePartAreas, hasTemplateParts, isLoading } =
 		useTemplatePartAreas();
 	const { patternCategories, hasPatterns } = usePatternCategories();
+	const isBlockBasedTheme = useSelect(
+		( select ) => select( coreStore ).getCurrentTheme()?.is_block_theme,
+		[]
+	);
+	const isTemplatePartsMode = useSelect( ( select ) => {
+		const settings = select( editSiteStore ).getSettings();
+		return !! settings.supportsTemplatePartsMode;
+	}, [] );
+	const templatePartsLink = useLink( {
+		path: '/wp_template_part/all',
+		// If a classic theme that supports template parts accessed
+		// the Patterns page directly, preserve that state in the URL.
+		didAccessPatternsPage: isTemplatePartsMode ? 1 : undefined,
```

Comment:
Nit: I know `isTemplatePartsMode` implicitly implies `! isBlockBasedTheme` but to me it's not that obvious from the naming alone 😅 . Do you think we should add `! isBlockBasedTheme` just to make it clearer?

Reply:
When someone comes to this fresh without much context, I think it could really help to be explicit here and probably doesn't hurt much to make this tweak.

Reply:
I see, added by b3963ef 👍